src/hotspot/share/prims/jvmtiEnvThreadState.cpp

```diff
@@ -241,16 +248,21 @@ void JvmtiEnvThreadState::set_frame_pop(int frame_number) {
 
 
 void JvmtiEnvThreadState::clear_frame_pop(int frame_number) {
+  assert(get_thread() == nullptr || get_thread()->is_handshake_safe_for(Thread::current()),
+         "frame pop data only accessible from same or detached thread or direct handshake");
+  JvmtiFramePop fpop(frame_number);
+  JvmtiEventController::clear_frame_pop(this, fpop);
+}
+
+void JvmtiEnvThreadState::clear_all_frame_pops() {
 #ifdef ASSERT
   Thread *current = Thread::current();
 #endif
   assert(get_thread() == nullptr || get_thread()->is_handshake_safe_for(current),
```

Comment:
```suggestion
  assert(get_thread() == nullptr || get_thread()->is_handshake_safe_for(Thread::current()),
```

Reply:
Thank you for the comment. Chris suggested the same.
I've fixed all such places in the file to make it consistent.